### DISCUSSION TITLE
question about the algorithm of solving the cyclic tridiagnal system (i.e. when periodic conditions are used)

### DISCUSSION THREAD

**Discussion by esaka-forever** [xcompact3d/Incompact3d#257] [Q&A] [answered]
Dear community, can anybody tell me what algorithm has been used in solving the cyclic tridiagnal system (which is used in the filtering subroutine and also for the calculation of the derivatives)? Or is there a reference for the algorithm used in Xcompact3d? Any help is appreciated !!!

**Comment by slaizet** [comment, marked as answer]
Hi,
We are using the Thomas algorithm.
Best,
Sylvain

**Reply by esaka-forever** [reply]
Thank you !
